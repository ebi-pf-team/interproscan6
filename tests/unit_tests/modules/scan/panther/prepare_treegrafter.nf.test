nextflow_process {

    name "Test Process PREPARE_TREEGRAFTER"
    script "modules/panther/main.nf"
    process "PREPARE_TREEGRAFTER"

    stage {
        symlink "tests/data/channels/panther.hmmsearch.out"
        symlink "tests/data/databases/panther/19.0"
    }

    test("Should run PREPARE_SMART without failures") {

        when {
            process {
                """
                input[0] = [1, file("meta/tests/data/channels/panther.hmmsearch.out")]
                input[1] = "meta/tests/data/databases/panther/19.0/"
                input[2] = "Tree_MSF"
                """
            }
        }

        then {
            assert process.success
        }
    }
}
