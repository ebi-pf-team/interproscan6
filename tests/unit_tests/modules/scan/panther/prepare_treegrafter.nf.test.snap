{
    "Should run PREPARE_SMART without failures": {
        "content": [
            {
                "0": [
                    [
                        1,
                        "panther.json:md5,7499c2af21f6cb70371cd76b251fc951"
                    ]
                ],
                "1": [
                    [
                        1,
                        [
                            "E8872C7A0261B9E88E6FF47EB34E4162"
                        ],
                        [
                            "PTHR13693"
                        ],
                        [
                            "tmp14335088482561019285.fa:md5,6a1e11e8a9ca556a8b6c051378f9b4f2"
                        ]
                    ]
                ],
                "fasta": [
                    [
                        1,
                        [
                            "E8872C7A0261B9E88E6FF47EB34E4162"
                        ],
                        [
                            "PTHR13693"
                        ],
                        [
                            "tmp14335088482561019285.fa:md5,6a1e11e8a9ca556a8b6c051378f9b4f2"
                        ]
                    ]
                ],
                "json": [
                    [
                        1,
                        "panther.json:md5,7499c2af21f6cb70371cd76b251fc951"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-06-02T13:58:08.532039"
    }
}